aptos-move/framework/aptos-framework/sources/delegation_pool.move

```diff
@@ -0,0 +1,1719 @@
+        });
+
+        // save delegation pool ownership and resource account address (inner stake pool address) on `owner`
+        move_to(owner, DelegationPoolOwnership { pool_address });
+    }
+
+    #[view]
+    public fun owner_cap_exists(addr: address): bool {
+        exists<DelegationPoolOwnership>(addr)
+    }
+
+    fun assert_owner_cap_exists(owner: address) {
+        assert!(owner_cap_exists(owner), error::not_found(EOWNER_CAP_NOT_FOUND));
+    }
+
+    #[view]
+    public fun get_owned_pool_address(owner: address): address acquires DelegationPoolOwnership {
+        assert_owner_cap_exists(owner);
+        borrow_global<DelegationPoolOwnership>(owner).pool_address
+    }
+
+    #[view]
+    public fun delegation_pool_exists(addr: address): bool {
+        exists<DelegationPool>(addr)
+    }
+
+    fun assert_delegation_pool_exists(pool_address: address) {
+        assert!(delegation_pool_exists(pool_address), error::invalid_argument(EDELEGATION_POOL_DOES_NOT_EXIST));
+    }
+
+    /// Retrieves the shared resource account owning the stake pool in order
+    /// to forward a stake-management operation to this underlying pool.
+    fun retrieve_stake_pool_signer(pool: &DelegationPool): signer {
+        account::create_signer_with_capability(&pool.stake_pool_signer_cap)
+    }
+
+    /// Get the address of delegation pool reference `pool`.
+    fun get_pool_address(pool: &DelegationPool): address {
+        account::get_signer_capability_address(&pool.stake_pool_signer_cap)
+    }
+
+    #[view]
+    /// Return the index of current observed lockup cycle on delegation pool `pool_address`.
+    /// This represents the key into `inactive_shares` of the `pending_inactive` shares pool.
+    public fun observed_lockup_cycle(pool_address: address): u64 acquires DelegationPool {
+        assert_delegation_pool_exists(pool_address);
+        borrow_global<DelegationPool>(pool_address).observed_lockup_cycle
+    }
+
+    /// Allows an owner to change the operator of the underlying stake pool.
+    public entry fun set_operator(owner: &signer, new_operator: address) acquires DelegationPoolOwnership, DelegationPool {
+        let pool_address = get_owned_pool_address(signer::address_of(owner));
+        // synchronize delegation and stake pools before any user operation
+        synchronize_delegation_pool(pool_address);
+        stake::set_operator(&retrieve_stake_pool_signer(borrow_global<DelegationPool>(pool_address)), new_operator);
+    }
+
+    /// Allows an owner to change the delegated voter of the underlying stake pool.
+    public entry fun set_delegated_voter(owner: &signer, new_voter: address) acquires DelegationPoolOwnership, DelegationPool {
+        let pool_address = get_owned_pool_address(signer::address_of(owner));
+        // synchronize delegation and stake pools before any user operation
+        synchronize_delegation_pool(pool_address);
```

Comment:
This is likely for the overall benefit of keeping things updated for this delegation pool but at the cost of increasing gas cost for set_delegated_voter. What do you think about omitting the sync here?

Reply:
you're right, sync is called here only for uniformity, it doesn't introduce any inconsistency if omitted

Reply:
We should just keep this for consistency since it's called in every entry function.